overlord/hookstate/ctlcmd/helpers.go

```diff
@@ -80,12 +92,19 @@ func runServiceCommand(context *hookstate.Context, inst *servicestate.Instructio
 		return err
 	}
 
-	chg, err := servicechangeImpl(st, appInfos, inst)
+	ts, err := servicechangeImpl(st, appInfos, inst)
 	if err != nil {
 		return err
 	}
 
+	if !context.IsEphemeral() && context.HookName() == "configure" && (inst.Action == "restart" || inst.Action == "start") {
```

Comment:
Okay, you are checking for IsEphemeral here. Still, this is very dangerous. There's zero hint in this whole file that this would panic.

Also, why those actions only? What about stop? Any others?

Reply:
Right. I deliberately left `stop` out assuming it's not interested in any values since the service is just going to be stopped. But we actually may need to queue it to preserve the order of operations (the hook may execute multitple commands, sequence matters)... So yes, I think `stop` needs to be queued too. Thanks.